internal/parser/parser.go

```diff
@@ -246,7 +247,40 @@ func (p *parser) parseQuery() ast.Query {
 		p.error(pos, "no comment preceding query")
 		return &ast.BadQuery{From: pos, To: p.pos}
 	}
-	last := doc.List[len(doc.List)-1]
+
+	paramTypeOverrides := make(map[string]string, 4)
+	annotationPos := len(doc.List) - 1
+
+	for ; annotationPos > 0; annotationPos-- {
```

Comment:
I was taking insipiration from how the `name: :kind` annotation was parsed but I agree it is a bit messy and can be improved. (N.B.)I've moved the parsing of go types into it's own function to clean things up already. I just need to push it.

Can you elaborate on what you mean by proper parser? Do you want a custom built DSL and parser or do you have something in mind to be reused?  
One idea would be to find the `name:` line, then parse everything that's afterwards in a known language like TOML (and we can enforce the structure of it). This would have the advantage to reusue a known configuration language and libs to parse it. However while it's close to the CLI it is a bit different, notably spaces around the equal are allowed and the values (i.e. go types) have to be quoted.

Which way would you like to go?

Reply:
To give an example of what I mean with TOML, it would work like this:
```SQL
-- name: GetDailyFlightsFromAircraft :many
-- [arg]
-- day = "github.com/0xjac/custom-project/types.Day"
-- [return]
-- departure = "time.Time"
-- eta = "*time.Time"
SELECT f.flight_number, f.departure, f.arrival, f.eta
FROM flights f
WHERE pggen.arg("day") <= f.departure AND f.departure < pggen.arg("day") + 1
ORDER BY f.departure DESC;
```